tests/testing.rs

```diff
@@ -21,3 +29,33 @@ fn mdbook_detects_book_with_failing_tests() {
 
     assert!(md.test(vec![]).is_err());
 }
+
+#[test]
+fn mdbook_runs_preprocessors() {
+
+    lazy_static! {
+        static ref HAS_RUN: Mutex<bool> = Mutex::new(false);
```

Comment:
I don't think we need to use `lazy_static!()` here, an `Arc<Mutex<bool>>` should be enough.

Reply:
You're right, I was trying to have a `dummy` option in the Config and have the book add the `DummyPreprocessor` to the list when it found the `dummy` option, which did require the `lazy_static` since the test wasn't constructing it directly. I should've rethought the design when I switched to using `with_preprocessor` and building the `DummyPreprocessor` directly.